R/textstat_simil.R

```diff
@@ -622,6 +624,8 @@ textstat_proxy <- function(x, y = NULL,
     }
 
     margin <- match.arg(margin)
+
+    if (method == "hamman") method <- "hamann" # trap older "hamman" spelling
```

Comment:
When I was fixing proxyC, I noticed that `method[method == "hamman"] <- "hamann"` works better, because `method` can be a vector.
```
Warning message:
In if (method == "hamman") method <- "hamann" :
  the condition has length > 1 and only the first element will be used
```

Reply:
I thought of that, but we cannot allow vectors for `method` since the result could then not be a (single) matrix. Seems to be the same in **proxyC**:

``` r
mt <- Matrix::rsparsematrix(5, 5, 0.3)
proxyC::simil(mt, method = c("cosine", "euclidean"))
## Error in match.arg(method): 'arg' must be of length 1
```

Reply:
If `method` is unspecified by the user, it is a vector before `match.arg()`. 
```
> test <- function(x = c("a", "b", "c")) {
+     print(x)
+     x <- match.arg(x)
+     print(x)
+ }
> test("b")
[1] "b"
[1] "b"
> test()
[1] "a" "b" "c"
[1] "a"
``` 
See https://github.com/koheiw/proxyC/blob/73c33074ba9a7f1b96533749479ad77aafbbf1b8/R/proxy.R#L79.

Reply:
Done